libraries/app/database_api.cpp

```diff
@@ -2061,48 +2061,41 @@ vector<blinded_balance_object> database_api_impl::get_blinded_balances( const fl
 //                                                                  //
 //////////////////////////////////////////////////////////////////////
 
-vector<withdraw_permission_object> database_api::get_withdrawals_giver(account_id_type account, uint32_t start, uint32_t limit)const
+vector<withdraw_permission_object> database_api::get_withdrawals_giver(account_id_type account, object_id_type start, uint32_t limit)const
 {
    return my->get_withdrawals_giver( account, start, limit );
 }
 
-vector<withdraw_permission_object> database_api_impl::get_withdrawals_giver(account_id_type account, uint32_t start, uint32_t limit)const
+vector<withdraw_permission_object> database_api_impl::get_withdrawals_giver(account_id_type account, object_id_type start, uint32_t limit)const
 {
    FC_ASSERT( limit <= 100 );
    vector<withdraw_permission_object> result;
 
-   auto withdraw_range = _db.get_index_type<withdraw_permission_index>().indices().get<by_from>().equal_range(account);
-   auto total = distance(withdraw_range.first, withdraw_range.second);
-   if(start >= total ) return result;
-   advance(withdraw_range.first, start);
-
-   std::for_each(withdraw_range.first, withdraw_range.second,
-                 [&result] (const withdraw_permission_object& withdraw) {
-                    result.emplace_back(withdraw);
-                 });
-
+   auto withdraw_itr = _db.get_index_type<withdraw_permission_index>().indices().get<by_from>().lower_bound(boost::make_tuple(account, start));
```

Comment:
Perhaps use `upper_bound` for pagination, otherwise we will have effectively 99 rows per page, which will impact performance somehow.

Reply:
make sense change to `upper_bound` but i had to add a special case to be able to retrieve first object. this will not make too much sense in the live change as 1.12.0 is already gone but as i am in a private testnet i think the right thing is to consider it.